src/vs/workbench/browser/parts/quickopen/quickOpenController.ts

```diff
@@ -549,6 +568,33 @@ export class QuickOpenController extends WorkbenchComponent implements IQuickOpe
 		return promiseCompletedOnHide;
 	}
 
+	private handleOnCancel(isPicker: boolean): void {
+		// restore the editor part state after cancelling
+		this.historyService.block(true);
+
+		// restore the previous preview editor
+		if (this.previousPreviewEditorInput) {
```

Comment:
> Also you have to be careful that you do this from the onCancel() callback. Because as far as I remember this method is not called when you actually pick an entry from quick open. I would expect some additional logic when an element is picked.

I'll handle this in `handleOnHide` and use the `reason` arg to apply logic.